### PR TITLE
ISV-2928 - Forward logs to Splunk from community nginx proxy

### PR DESCRIPTION
**Pull request by mgreczi**

### CODE REVIEW

File: ansible/inventory/group_vars/operator-pipeline-integration-tests.yml [hunk 1]
```diff
@@ -20,6 +20,10 @@ community_signing_pipeline_url: "{{ community_signing_pipeline_name }}-{{ oc_nam
 
 nginx_proxy_url: "{{ community_signing_pipeline_name }}-nginx-{{ oc_namespace }}.apps.pipelines-stage.0ce8.p1.openshiftapps.com"
 
+log_dir: /var/log/nginx
```
Comment (mantomas) [inline ansible/inventory/group_vars/operator-pipeline-integration-tests.yml:23]: Do we need to copy this to multiple inventories, if there is no change from operator-pipeline.yaml?

File: ansible/roles/nginx-proxy/tasks/main.yml [hunk 2]
```diff
@@ -51,3 +51,10 @@
       - env={{ env }}
       - suffix!={{ suffix }}
   include_tasks: remove-objects.yml
+
+- name: Create Fluentd config
+  k8s:
+    state: present
+    force: yes
+    namespace: '{{ oc_namespace }}'
+    definition: "{{ lookup('template', '../templates/openshift/fluentd-config.yml') }}"mgreczi-mac:operator-pipelines
```
Comment (mporrato) [inline ansible/roles/nginx-proxy/tasks/main.yml:60]:
```suggestion
    definition: "{{ lookup('template', '../templates/openshift/fluentd-config.yml') }}"
```

File: ansible/roles/nginx-proxy/templates/openshift/nginx-configuration.yml [hunk 3]
```diff
@@ -20,18 +20,23 @@ data:
...
       server {
 
           listen 8080;
 
-          # enable compression
-          gzip on;
-          gzip_types text/plain application/json;
-          gzip_min_length 1024;
+	        access_log /var/log/nginx/access.log logger-json;
+	        error_log  /var/log/nginx/error.log logger-json;
```
Comment (mporrato) [inline ansible/roles/nginx-proxy/templates/openshift/nginx-configuration.yml:34]: The file names do not match the names used in the fluentd config.
Here you are configuring nginx to log in json format but, IIUC, you are telling fluentd to parse logs in the "traditional" combined format.

File: ansible/inventory/group_vars/operator-pipeline-integration-tests.yml (hunk 1 above)
Comment (Allda) [inline ansible/inventory/group_vars/operator-pipeline-integration-tests.yml:23]: You can define these vars in the main inventory and de-duplicate them.

File: ansible/roles/nginx-proxy/templates/openshift/nginx-configuration.yml [hunk 4]
```diff
@@ -20,18 +20,23 @@ data:
...
+             "resp_time": $request_time, "upstream_addr": "$upstream_addr"}';
 
       server {
 
           listen 8080;
 
-          # enable compression
-          gzip on;
-          gzip_types text/plain application/json;
-          gzip_min_length 1024;
```
Comment (mporrato) [inline ansible/roles/nginx-proxy/templates/openshift/nginx-configuration.yml:34]: I think we can keep response compression enabled.

File: ansible/roles/nginx-proxy/tasks/main.yml (hunk 2 above)
Comment (Allda) [inline ansible/roles/nginx-proxy/tasks/main.yml:60]: The end of this line doesn't look right.

File: ansible/roles/nginx-proxy/templates/openshift/fluentd-config.yml [hunk 5]
```diff
@@ -0,0 +1,73 @@
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: 'loki-fluentd-{{ suffix }}'
```
Comment (Allda) [inline ansible/roles/nginx-proxy/templates/openshift/fluentd-config.yml:5]: Please rename `loki` to `nginx` here and in the rest of the file.

File: ansible/roles/nginx-proxy/templates/openshift/nginx-configuration.yml [hunk 6]
```diff
@@ -20,18 +20,23 @@ data:
...
+					   "resp_body_size": $body_bytes_sent, "host": "$http_host", "address": "$remote_addr", \
+					   "request_length": $request_length, "method": "$request_method", "uri": "$request_uri", \
+             "status": $status,  "user_agent": "$http_user_agent", \
+             "resp_time": $request_time, "upstream_addr": "$upstream_addr"}';
 
       server {
 
           listen 8080;
 
-          # enable compression
```
Comment (Allda) [inline ansible/roles/nginx-proxy/templates/openshift/nginx-configuration.yml:31]: Is there any reason of removing a compression?

File: ansible/roles/nginx-proxy/templates/openshift/nginx-configuration.yml [hunk 7]
```diff
@@ -20,18 +20,23 @@ data:
...
-          gzip on;
-          gzip_types text/plain application/json;
-          gzip_min_length 1024;
+	        access_log /var/log/nginx/access.log logger-json;
```
Comment (Allda) [inline ansible/roles/nginx-proxy/templates/openshift/nginx-configuration.yml:33]: Please make line indentation consistent.

File: ansible/roles/operator-pipeline/defaults/main.yml [hunk 8]
```diff
@@ -7,3 +7,8 @@ oc_namespace: null
 suffix: unknown
 namespace_state: present
 github_webhook_state: present
+
+#### Fluentd settings
+splunk_hec_token: null
+splunk_hec: splunk-hec.corp.redhat.com
```
Comment (Allda) [inline ansible/roles/operator-pipeline/defaults/main.yml:13]: This should be defined in inventory.

File: ansible/inventory/group_vars/operator-pipeline-integration-tests.yml (hunk 1 above)
Comment (mgreczi) [reply ansible/inventory/group_vars/operator-pipeline-integration-tests.yml]: Corrected.

File: ansible/roles/nginx-proxy/templates/openshift/nginx-configuration.yml (hunk 4 above)
Comment (mgreczi) [reply ansible/roles/nginx-proxy/templates/openshift/nginx-configuration.yml]: I replaced.

File: ansible/roles/nginx-proxy/templates/openshift/nginx-configuration.yml (hunk 6 above)
Comment (mgreczi) [reply ansible/roles/nginx-proxy/templates/openshift/nginx-configuration.yml:31]: I had misconception of the role of this setting. I replaced.

File: ansible/inventory/group_vars/clusters.yml [hunk 9]
```diff
@@ -2,3 +2,8 @@ ansible_connection: local
 
 # Index signature verification vars
 registry_auth_local_path: ../../vaults/index-signature-verification/registry-auth.json
+
```
Comment (Allda) [inline ansible/inventory/group_vars/clusters.yml:5]: This should go into operator-pipeline.yml group vars.

File: ansible/roles/nginx-proxy/tasks/main.yml [hunk 10]
```diff
@@ -51,3 +51,10 @@
       - env={{ env }}
       - suffix!={{ suffix }}
   include_tasks: remove-objects.yml
+
```
Comment (Allda) [inline ansible/roles/nginx-proxy/tasks/main.yml:54]: The config should be created before a nginx is deployed. Please rearrange the order of tasks.

File: ansible/roles/nginx-proxy/templates/openshift/fluentd-config.yml [hunk 11]
```diff
@@ -0,0 +1,73 @@
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: '{{ community_signing_pipeline_name }}-fluentd-{{ suffix }}'
+  labels:
+    io.kompose.service: "{{ community_signing_pipeline_name }}"
+    app: API-key-server
```
Comment (Allda) [inline ansible/roles/nginx-proxy/templates/openshift/fluentd-config.yml:8]: The app label is still wrong pointing to a Loki app again.

File: ansible/roles/nginx-proxy/templates/openshift/nginx-deployment.yml [hunk 12]
```diff
@@ -71,3 +74,30 @@ spec:
...
+          image: images.paas.redhat.com/exd-sp-guild-isv/fluentd:latest
+          env:
+            - name: FLUENTD_CONF
+              value: fluentd.conf
+          volumeMounts:
+            - name: "{{ community_signing_pipeline_name }}-logs"
+              mountPath: /var/log/nginx
+              readOnly: true
+            - name: fluentd-config
+              mountPath: /fluentd/etc
```
Comment (mporrato) [inline ansible/roles/nginx-proxy/templates/openshift/nginx-deployment.yml:87]: perhaps you meant

```suggestion
              mountPath: /etc/fluentd
```

File: ansible/roles/nginx-proxy/templates/openshift/nginx-configuration.yml [hunk 13]
```diff
@@ -20,21 +20,33 @@ data:
 
     http {
 
-      log_format compression '$remote_addr - $remote_user [$time_local] '
-                           '"$request" $status $body_bytes_sent '
-                           '"$http_referer" "$http_user_agent" "$gzip_ratio"';
+      log_format logger-json escape=json '{"source": "nginx", "time": $msec, \
```
Comment (mantomas) [inline ansible/roles/nginx-proxy/templates/openshift/nginx-configuration.yml:23]: Is the source "nginx"? Or should this be "operator-pipeline" and "nginx" the name?

**Review by mporrato — APPROVED** [review]

**Review by Allda — APPROVED** [review]

**Pull request merged**